### PR TITLE
feat(auth): remove the APPLICATION_URL parameter

### PR DESCRIPTION
**Pull request by vamonte**
the callback_url is now useless in the oic registering. application is now useless too

### General

- [ ] Write an explicit title for the Pull Request, following [Conventional Commits specification](https://www.conventionalcommits.org)
- [ ] Test manually the implemented changes
- [ ] Validate the code quality (indentation, syntax, style, simplicity, readability)

### Security

- [ ] Consider the security impact of the changes made

### CODE REVIEW

**Review by Thenkei — COMMENTED** [review]
- ✅  Code review - You nailed it! 🚀 I just have a few questions.
- ✅  Automatic tests
  - ✅  Unit tests
- 🕐 Manual tests - We can make a call to see this together in live? 🤩 
- ✅  PR title

File: django_forest/authentication/views/callback.py [hunk 1]
```diff
@@ -118,9 +118,8 @@ def _authenticate(self, client, state, code):
         else:
             self._handle_authent_error(response)
 
-    def _verify_code_and_generate_token_body(self, redirect_url, request):
-        client = OidcClientManager.get_client_for_callback_url(redirect_url)
-        client.do_access_token_request
```
[the remark this answers is not in the document]
Comment (vamonte) [reply django_forest/authentication/views/callback.py:123]: It should be a typo. `client.do_access_token_request` is a method but it wasn't called.

File: django_forest/authentication/oidc/client_manager.py [hunk 2]
```diff
@@ -31,14 +30,14 @@ def get_client_for_callback_url(cls, callback_url):
...
-            redirect_uri=callback_url,
+            redirect_uri=client_credentials['redirect_uris'][0],
             issuer=client_data['issuer'],
             authorization_endpoint=urljoin(client_data['issuer'], 'oidc/auth'),
             token_endpoint=urljoin(client_data['issuer'], 'oidc/token'),
             jwks_uri=urljoin(client_data['issuer'], 'oidc/jwks')
         )
-
         cls.client = Client(client_data['client_id'], verify_ssl=False)
+        cls.client.redirect_uris = client_credentials['redirect_uris']
```
[the remark this answers is not in the document]
Comment (vamonte) [reply django_forest/authentication/oidc/client_manager.py:40]: It is the way to retrieve the`redirect_uris `in the `_get_authorization_url` method.

**Review by Thenkei — APPROVED** [review]

File: django_forest/authentication/oidc/client_manager.py [hunk 3]
```diff
@@ -13,14 +13,13 @@ class OidcClientManager:
...
     @classmethod
-    def get_client_for_callback_url(cls, callback_url):
+    def get_client(cls):
         if cls.client:
             return cls.client
 
         configuration = retrieve()
         client_credentials = register({
             'token_endpoint_auth_method': 'none',
-            'redirect_uris': [callback_url],
```
Comment (Thenkei) [inline django_forest/authentication/oidc/client_manager.py:23]: @jbarreau This is important

File: django_forest/authentication/oidc/client_manager.py [hunk 4]
```diff
@@ -31,14 +30,14 @@ def get_client_for_callback_url(cls, callback_url):
 
         op_info = ProviderConfigurationResponse(
             client_id=client_data['client_id'],
-            redirect_uri=callback_url,
+            redirect_uri=client_credentials['redirect_uris'][0],
```
Comment (Thenkei) [inline django_forest/authentication/oidc/client_manager.py:33]: @jbarreau And this too.

**Pull request merged**